### ISSUE TITLE
java.lang.NoSuchMethodError: com.google.common.collect.Maps.immutableEnumMap(Ljava/util/Map;)Lcom/google/common/collect/ImmutableMap;

### ISSUE THREAD

**Issue by rohtakdev** [wro4j/wro4j#978]
Can you please help in identifying the issue? I am getting below error -

```
2015-09-11T20:13:51.799-0700|Warning: StandardWrapperValve[phuzzle]: Servlet.service() for servlet phuzzle threw exception
java.lang.NoSuchMethodError: com.google.common.collect.Maps.immutableEnumMap(Ljava/util/Map;)Lcom/google/common/collect/ImmutableMap;
	at com.google.javascript.jscomp.parsing.parser.Keywords.<clinit>(Keywords.java:94)
	at com.google.javascript.jscomp.parsing.parser.Scanner.scanIdentifierOrKeyword(Scanner.java:681)
	at com.google.javascript.jscomp.parsing.parser.Scanner.scanToken(Scanner.java:569)
	at com.google.javascript.jscomp.parsing.parser.Scanner.peekToken(Scanner.java:228)
	at com.google.javascript.jscomp.parsing.parser.Parser.peekToken(Parser.java:2886)
	at com.google.javascript.jscomp.parsing.parser.Parser.peekToken(Parser.java:2879)
	at com.google.javascript.jscomp.parsing.parser.Parser.getTreeStartLocation(Parser.java:2798)
	at com.google.javascript.jscomp.parsing.parser.Parser.parseProgram(Parser.java:154)
	at com.google.javascript.jscomp.parsing.ParserRunner.parse(ParserRunner.java:101)
	at com.google.javascript.jscomp.JsAst.parse(JsAst.java:82)
	at com.google.javascript.jscomp.JsAst.getAstRoot(JsAst.java:48)
	at com.google.javascript.jscomp.CompilerInput.getAstRoot(CompilerInput.java:113)
	at com.google.javascript.jscomp.Compiler.hoistExterns(Compiler.java:1456)
	at com.google.javascript.jscomp.Compiler.parseInputs(Compiler.java:1358)
	at com.google.javascript.jscomp.Compiler.parse(Compiler.java:784)
	at com.google.javascript.jscomp.Compiler.compileInternal(Compiler.java:745)
	at com.google.javascript.jscomp.Compiler.access$000(Compiler.java:94)
	at com.google.javascript.jscomp.Compiler$3.call(Compiler.java:657)
	at com.google.javascript.jscomp.Compiler$3.call(Compiler.java:654)
	at com.google.javascript.jscomp.Compiler.runInCompilerThread(Compiler.java:726)
	at com.google.javascript.jscomp.Compiler.compile(Compiler.java:654)
	at com.google.javascript.jscomp.Compiler.compile(Compiler.java:624)
	at ro.isdc.wro.extensions.processor.js.GoogleClosureCompressorProcessor.process(GoogleClosureCompressorProcessor.java:117)
	at ro.isdc.wro.model.resource.processor.decorator.ProcessorDecorator.process(ProcessorDecorator.java:86)
	at ro.isdc.wro.model.resource.processor.decorator.LazyProcessorDecorator.process(LazyProcessorDecorator.java:49)
	at ro.isdc.wro.model.resource.processor.decorator.ProcessorDecorator.process(ProcessorDecorator.java:86)
	at ro.isdc.wro.model.resource.processor.decorator.ProcessorDecorator.process(ProcessorDecorator.java:86)
	at ro.isdc.wro.model.resource.processor.decorator.ProcessorDecorator.process(ProcessorDecorator.java:86)
	at ro.isdc.wro.model.resource.processor.decorator.SupportAwareProcessorDecorator.process(SupportAwareProcessorDecorator.java:39)
	at ro.isdc.wro.model.resource.processor.decorator.ProcessorDecorator.process(ProcessorDecorator.java:86)
	at ro.isdc.wro.model.resource.processor.decorator.ExceptionHandlingProcessorDecorator.process(ExceptionHandlingProcessorDecorator.java:56)
	at ro.isdc.wro.model.resource.processor.decorator.ProcessorDecorator.process(ProcessorDecorator.java:86)
	at ro.isdc.wro.model.resource.processor.decorator.BenchmarkProcessorDecorator.process(BenchmarkProcessorDecorator.java:44)
	at ro.isdc.wro.model.resource.processor.decorator.ProcessorDecorator.process(ProcessorDecorator.java:86)
	at ro.isdc.wro.model.resource.processor.decorator.DefaultProcessorDecorator.process(DefaultProcessorDecorator.java:42)
	at ro.isdc.wro.model.group.processor.PreProcessorExecutor$2.process(PreProcessorExecutor.java:228)
	at ro.isdc.wro.model.group.processor.PreProcessorExecutor.applyPreProcessors(PreProcessorExecutor.java:207)
	at ro.isdc.wro.model.group.processor.PreProcessorExecutor.processAndMerge(PreProcessorExecutor.java:103)
	at ro.isdc.wro.model.group.processor.PreProcessorExecutor.processAndMerge(PreProcessorExecutor.java:77)
	at ro.isdc.wro.model.group.processor.GroupsProcessor.process(GroupsProcessor.java:83)
	at ro.isdc.wro.cache.support.DefaultSynchronizedCacheStrategyDecorator.loadValue(DefaultSynchronizedCacheStrategyDecorator.java:101)
	at ro.isdc.wro.cache.support.DefaultSynchronizedCacheStrategyDecorator.loadValue(DefaultSynchronizedCacheStrategyDecorator.java:35)
	at ro.isdc.wro.cache.support.AbstractSynchronizedCacheStrategyDecorator.get(AbstractSynchronizedCacheStrategyDecorator.java:55)
	at ro.isdc.wro.manager.ResourceBundleProcessor.serveProcessedBundle(ResourceBundleProcessor.java:63)
	at ro.isdc.wro.manager.WroManager.process(WroManager.java:159)
	at ro.isdc.wro.http.WroFilter.processRequest(WroFilter.java:335)
	at ro.isdc.wro.http.WroFilter.doFilter(WroFilter.java:289)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:256)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:316)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:160)
	at org.apache.catalina.core.StandardPipeline.doInvoke(StandardPipeline.java:734)
	at org.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:673)
	at com.sun.enterprise.web.WebPipeline.invoke(WebPipeline.java:99)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:174)
	at org.apache.catalina.connector.CoyoteAdapter.doService(CoyoteAdapter.java:415)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:282)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:459)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:167)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:201)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:175)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:235)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:561)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)
	at java.lang.Thread.run(Thread.java:745)

```

I am using below dependency -
```xml
	<dependency>
			<groupId>ro.isdc.wro4j</groupId>
			<artifactId>wro4j-core</artifactId>
			<version>1.7.8</version>
		</dependency>
		<dependency>
			<groupId>ro.isdc.wro4j</groupId>
			<artifactId>wro4j-extensions</artifactId>
			<version>1.7.8</version>
		</dependency>
		<dependency>
			<groupId>com.google.guava</groupId>
			<artifactId>guava</artifactId>
                        <version>18.0</version>
		</dependency>
```
 wro.xml -

```xml
<?xml version="1.0" encoding="UTF-8"?>
<groups xmlns="http://www.isdc.ro/wro">
  <group name="play">
    <js minimize="true">/resources/js/play.js</js>
  </group>
</groups> 
```
wro.properties

```properties
managerFactoryClassName=ro.isdc.wro.manager.factory.ConfigurableWroManagerFactory
preProcessors=googleClosureAdvanced
postProcessors=
uriLocators=servletContext,uri,classpath
# Configurable options available since 1.4.7 (not mandatory)
hashStrategy=MD5
namingStrategy=hashEncoder-CRC32
```

**Comment by alexo** [comment]
At the first glance, it seems that the google closure library requires a different version of guava. The one found in your classpath is incompatible. What happens if you let the google closure to retrieve the guava dependency transitively?

**Comment by alexo** [comment]
The immutableEnumMap method is available in guava version 18.0. 
The most probable cause for your exception is that you have an older version of the library in your classpath. Closing this issue with "invalid" resolution.
